cmake/spheral_cxx-config.cmake.in

```diff
@@ -4,9 +4,28 @@
 if(NOT axom_DIR)
   set(axom_DIR "@axom_DIR@" CACHE PATH "")
 endif()
-set(SPHERAL_BLT_DIR "@SPHERAL_BLT_DIR@")
 set(SPHERAL_CXX_INSTALL_PREFIX "@CMAKE_INSTALL_PREFIX@")
 set(SPHERAL_CXX_INCLUDE_DIRS "${SPHERAL_CXX_INSTALL_PREFIX}/include")
+if(NOT @ENABLE_CXXONLY@ AND NOT TARGET Python3)
+  set(python_DIR "@python_DIR@")
+  set(Python_EXECUTABLE ${python_DIR}/bin/python3)
+  set(Python3_EXECUTABLE ${python_DIR}/bin/python3)
+  find_package(Python3 COMPONENTS Interpreter Development)
+  set(PYTHON_EXE ${Python3_EXECUTABLE})
+endif()
+
+if(NOT DEFINED CMAKE_C_COMPILER)
+  set(CMAKE_C_COMPILER "@CMAKE_C_COMPILER@" CACHE PATH "Spheral C compiler path")
+  set(CMAKE_CXX_COMPILER "@CMAKE_CXX_COMPILER@" CACHE PATH "Spheral C++ compiler path")
+  set(CMAKE_Fortran_COMPILER "@CMAKE_Fortran_COMPILER@" CACHE PATH "Spheral C++ compiler path")
+endif()
+set(SPHERAL_ENABLE_MPI @ENABLE_MPI@ CACHE BOOL "")
+if(SPHERAL_ENABLE_MPI AND NOT DEFINED MPI_C_COMPILER)
+  set(MPI_C_COMPILER "@MPI_C_COMPILER@" CACHE PATH "")
+  set(MPI_CXX_COMPILER "@MPI_CXX_COMPILER@" CACHE PATH "")
+  set(MPI_Fortran_COMPILER "@MPI_Fortran_COMPILER@" CACHE PATH "")
```

Comment:
Is this going to set the compilers for every project that tries to include Spheral in it's CMake system? I think we might want to prefix these variable names with "SPHERAL_" so we aren't unintentionally forcing projects to use our compiler and MPI

Reply:
That was a concern of mine. I originally had it prefixed with SPHERAL but that didn't work because I think those flags needed to be set before spheral_cxx-targets is included. I hoped that guarding the sets with checks for whether they are already defined would be enough to prevent issues for other projects that already have those variables set. That was one of the things I hope Paul can try out.